### PR TITLE
Bump A-S to 0.27.0, expose bookmark syncing and history type filtering APIs

### PR DESCRIPTION
**Pull request by grigoryk**
This hackishly exposes the bookmark syncing APIs. We'll update this once appservices gets a mature sync manager.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt [hunk 1]
```diff
@@ -146,4 +149,21 @@ open class PlacesHistoryStorage(context: Context) : PlacesStorage(context), Hist
...
+     * @param authInfo The authentication information to sync with.
+     * @return Sync status of OK or Error
+     */
+    override suspend fun sync(authInfo: AuthInfo): SyncStatus {
+        return try {
+            withContext(scope.coroutineContext) {
+                places.syncHistory(authInfo.into())
+                SyncStatus.Ok
+            }
+        } catch (e: PlacesException) {
```
Comment (thomcc) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt]: Nit: could you report (or fail to catch, assuming something else catches and reports) rust panics here (and above at the bookmark sync)

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Connection.kt [hunk 2]
```diff
@@ -62,9 +66,14 @@ internal object RustPlacesConnection : Connection {
...
     }
 
-    override fun sync(syncInfo: SyncAuthInfo) {
+    override fun syncHistory(syncInfo: SyncAuthInfo) {
+        check(api != null) { "must call init first" }
+        api!!.syncHistory(syncInfo)
+    }
+
+    override fun syncBookmarks(syncInfo: SyncAuthInfo) {
         check(api != null) { "must call init first" }
```
Comment (thomcc) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Connection.kt]: Unrelated to absolutely anything, is `check` standard? Dang, we should be using it in a-s...

**Comment by grigoryk** [comment]
Currently blocked on getting 0.27.0 release mirrored. See https://bugzilla.mozilla.org/show_bug.cgi?id=1546211

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt (hunk 1 above)
Comment (grigoryk) [reply components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt]: This will be reported upstream.

**Pull request merged**